app/controllers/articles_controller.rb

```diff
@@ -185,6 +185,21 @@ def stats
     @organization_id = @article.organization_id
   end
 
+  def admin_unpublish
+    authorize @article
+    if @article.has_frontmatter?
+      @article.body_markdown.sub!("\npublished: true\n", "\npublished: false\n")
```

Comment:
This regexp will fail if `published:true`, or `published: true ` or other combinations that are valid Markdown frontamtters.

I think we should replace it with the following:


```suggestion
      @article.body_markdown.sub!(%r{\npublished:\s*true\s*\n}, "\npublished: false\n")
```

this takes into account optional spaces between `published:` and `true` and after `true` and before the `newline`